labs/a3.md

```diff
@@ -0,0 +1,520 @@
+the kernel command line. Once we're editing this command line, hit
+`END` to jump to the end of the line and carefully add a space and
+then `console=ttyS0`. When you're confident you've typed it right, hit
+enter to start booting the kernel.
+
+This interface we are using is basically the output of a virtual
+serial connection to the virtual machine. If you wanted to detatch
+from this, you would hit the key combo `ctrl-]`, and reconnect by
+running the command `sudo virsh console archvm`.
+
+> In most cases if you disconnect and have to restart in the middle,
+> you can just run `sudo virsh console archvm` to reconnect. If you
+> really mess up, you can start things over by stopping the VM
+> (`sudo virsh destroy archvm`), deleting the VM
+> (`sudo virsh undefine archvm --nvram --remove-all-storage`), and
+> then recreating it by starting over from the command above. If for
+> some reason, your VM gets turned off before you are done installing
+> Arch, it may be possible to reconnect the archiso and reboot the VM
+> using the archiso. However, doing this is out of scope for this lab.
+
+You should start seeing a bunch of lines that start with `[   OK   ]`,
+and eventually you will see a login prompt.
+
+> Since the installation media is basically just a preinstalled Linux
+> system that makes it easy to install Linux on another system, this
+> is the info about the booting "archiso" system. So, nothing is
+> actually installed on the VM yet, but we are booting into a
+> temporary Linux system that will help us install Linux permanently
+> on the VM.
+
+Once you see a login prompt, enter `root` as the username, and you
+should get in! You're now at the archiso command line.
+
+**GRADESCOPE**: Use the `hostname` command to get the current
+hostname. What is it? Then, use `ip addr` to find the current IP
+address (look at the `inet` line under `ens2`, it's the stuff at the
+beginning of the line, not including the `/24`). What is your IP
+address? In another terminal, try to ping this IP address from your
+student VM, and then also from tsunami (ssh.ocf.berkeley.edu). Can
+either of them reach it?
+
+> We'll discuss networking, and why you see these results, in a later
+> lecture.
+
+> Notice we are booted using UEFI (hence `-boot uefi` in the
+> virt-install line). Check out the ArchWiki
+> ([https://wiki.archlinux.org/index.php/Installation_guide#Verify_the_boot_mode](https://wiki.archlinux.org/index.php/Installation_guide#Verify_the_boot_mode))
+> for more info. Also, notice that we are already connected to the
+> internet! Our virtual network uses DHCP, and the archiso knows to
+> try to autoconfigure using DHCP. We will talk more about DHCP and
+> other networking configuration later in the course. Finally, notice
+> that we've conveniently inherited the system clock, so the time
+> inside the VM should be right. You can run `date` to see.
+
+## partitioning time
+
+The first order of business is to set up the partitions that we will
```

Comment:
Explain what partitions are and why we use them

Reply:
We spend a _lot_ of time in lecture on this, and I don't really think it will be valuable at that point.

Reply:
fair enough, i didn't read the lecture yet